joblib/parallel.py

```diff
@@ -576,6 +576,9 @@ def _effective_n_jobs(self):
     def _terminate_backend(self):
         if self._backend is not None:
             self._backend.terminate()
+            # reset backend when done,
+            # see https://github.com/joblib/joblib/issues/533
+            self._backend.__dict__ = {}
```

Comment:
We should not change the internal state of the backends using such private implementation from the outside like that.

It's the responsibility of the particular backend implementations to reset there states in terminate. In particular only the backend that maintain a `_effective_batch_size` attribute should reset there batching statistics to fix the original issue reported in #533.

Reply:
Indeed, the problem was related to batching statistics. I updated the PR accordingly.